unit/adsr.go

```diff
@@ -0,0 +1,149 @@
+package unit
+
+import (
+	"math"
+
+	"buddin.us/shaden/dsp"
+)
+
+func newAdsr(name string, _ Config) (*Unit, error) {
+	io := NewIO()
+	return NewUnit(io, name, &adsr{
+		state: &adsrState{
+			lastTrigger: -1,
+		},
+		stateFunc:  adsrIdle,
+		trigger:    io.NewIn("trigger", dsp.Float64(0)),
+		gate:       io.NewIn("gate", dsp.Float64(0)),
+		attack:     io.NewIn("attack", dsp.Duration(50)),
+		decay:      io.NewIn("decay", dsp.Duration(50)),
+		sustain:    io.NewIn("sustain", dsp.Duration(50)),
```

Comment:
What do you think about renaming this to `sustain-hold` and the `sustain-level` to `sustain`? Sustain in more conventional terms would indicate the level for anyone coming from other synths. I would also consider just `hold`, but because AHDSR envelopes exist I wonder if that would be confused with `hold`s meaning there.

Reply:
Yes, you are right!